### PR TITLE
Avoid adding deps every setup

### PR DESCRIPTION
**Pull request by mtsokol**
This PR makes sure that `Finch` dependency is added only once. This should improve startup time (both locally and in benchmarks).

### CODE REVIEW

File: src/finch/julia.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
 import juliapkg
 
-juliapkg.add("Finch", "9177782c-1635-4eb9-9bfb-d9dfa25e6bce", version="0.6.20")
+deps = juliapkg.deps.load_cur_deps()
+if not (
+    "packages" in deps and
+    "Finch" in deps["packages"] and
+    deps["packages"]["Finch"]["version"] == "0.6.20"
```
Comment (hameerabbasi) [inline src/finch/julia.py]: I'd simplify this to:
`deps.get(packages, {}).get("Finch", {}).get("version", None) != "0.6.20"`.

Comment (mtsokol) [reply src/finch/julia.py]: Done!

File: src/finch/julia.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import juliapkg
 
-juliapkg.add("Finch", "9177782c-1635-4eb9-9bfb-d9dfa25e6bce", version="0.6.20")
+deps = juliapkg.deps.load_cur_deps()
+if deps.get("packages", {}).get("Finch", {}).get("version", None) == "0.6.20":
```
Comment (hameerabbasi) [inline src/finch/julia.py]:
```suggestion
if deps.get("packages", {}).get("Finch", {}).get("version", None) != "0.6.20":
```

File: src/finch/julia.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import juliapkg
 
-juliapkg.add("Finch", "9177782c-1635-4eb9-9bfb-d9dfa25e6bce", version="0.6.20")
+deps = juliapkg.deps.load_cur_deps()
+if not deps.get("packages", {}).get("Finch", {}).get("version", None) == "0.6.20":
+    juliapkg.add("Finch", "9177782c-1635-4eb9-9bfb-d9dfa25e6bce", version="0.6.20")
```
Comment (hameerabbasi) [inline src/finch/julia.py]: I'd define `_FINCH_VERSION` and `_FINCH_HASH` and use those -- so it doesn't need updating in two places.

File: src/finch/julia.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import juliapkg
 
-juliapkg.add("Finch", "9177782c-1635-4eb9-9bfb-d9dfa25e6bce", version="0.6.20")
+deps = juliapkg.deps.load_cur_deps()
+if not deps.get("packages", {}).get("Finch", {}).get("version", None) == "0.6.20":
```
Comment (hameerabbasi) [inline src/finch/julia.py]:
```suggestion
if deps.get("packages", {}).get("Finch", {}).get("version", None) != "0.6.20":
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Last changes from my side. 😉

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
Lgtm! Did this end up improving the runtime? The pkg add should already be caching what it can, so I'm curious if this is noticeably faster.

**Comment by mtsokol** [comment]
The startup is faster for sure - `import finch` does run faster if it has already been imported with a given version.

**Comment by willow-ahrens** [comment]
awesome!

**Pull request merged**